exams/2006/pagrindinis/u2/u2.py

```diff
@@ -0,0 +1,80 @@
+DUOMENU_FAILAS = "Duom2.txt"
+NAUJOS_EILUTĖS_SIMBOLIS = "\n"
+RAŠYMO_REŽIMAS = 'w'
+SKAITYMO_REŽIMAS = 'r'
+TEKSTO_KODUOTĖ = 'utf-8'
+SIMBOLIŲ_KIEKIS_MIESTO_PAVADINIMUI = 15
+REZULTATŲ_FAILAS = "Rez2.txt"
+MINUTĖS_VALANDOJE = 60
+
+
+def nuskaityti_duomenis_iš_failo(duomenu_failas):
+    failas = open(duomenu_failas, SKAITYMO_REŽIMAS, encoding=TEKSTO_KODUOTĖ)
+    duomenys = failas.read()
+    failas.close()
+    return duomenys
+
+
+def apskaičiuoti_sustojimų_laikus(duomenų_failo_tekstas):
+    def laikas(atstumas, greitis):
+        return round(atstumas / greitis * MINUTĖS_VALANDOJE)
+
+    def apskaičiuoti_atvykimo_į_stotelę_laiką(
+        autobuso_vidutinis_greitis,
+        ankstesnio_sustojimo_valanda,
+        ankstesnio_sustojimo_minutės,
+        atstumas):
+
+        sustojimo_minutės = ankstesnio_sustojimo_minutės + laikas(atstumas, autobuso_vidutinis_greitis)
+
+        if sustojimo_minutės >= MINUTĖS_VALANDOJE:
+            sustojimo_valanda = ankstesnio_sustojimo_valanda + sustojimo_minutės // MINUTĖS_VALANDOJE
+            sustojimo_minutės = sustojimo_minutės % MINUTĖS_VALANDOJE
+        else:
+            sustojimo_valanda = ankstesnio_sustojimo_valanda
+
+        return sustojimo_valanda, sustojimo_minutės
+
+    # Suskaldome tekstą į masyvą, naudodami naujos eilutės simbolį \n, kaip skaldymo ženklą
+    teksto_eilučių_masyvas = duomenų_failo_tekstas.split(NAUJOS_EILUTĖS_SIMBOLIS)
```

Comment:
Šioje vietoje nėra reikalo naudoti `split`, yra toks dalykas kaip `splitlines`. O be to, žymiai efektyviau perduoti ne visą failo turinį nuskaitymą į atmintį, o tiesiog atidaryto failo objektą, kurį galima iteruoti per eilutes, tokiu atveju nereikia nei `split`, nei `splitlines`.

Reply:
Šiose užduotyse neverta svarstyti apie atminties naudojimo geriausias praktikas. Kad būtų paprasčiau, pakeisiu į list(failas) arba next(failas) pirmai eilutei, kur yra reikalingų duomenų, o po to list(failas), kad surinkti likusius interatoriaus duomenis.